src/Bridge/Symfony/Bundle/Test/Constraint/MatchesJsonSchema.php

```diff
@@ -46,15 +47,16 @@ public function toString(): string
     }
 
     /**
-     * @param array $other
+     * @param array|object $other
      */
     protected function matches($other): bool
     {
         if (!class_exists(Validator::class)) {
             throw new \RuntimeException('The "justinrainbow/json-schema" library must be installed to use "assertMatchesJsonSchema()". Try running "composer require --dev justinrainbow/json-schema".');
         }
 
-        $other = (object) $other;
+        $other = (string) json_encode($other);
+        $other = json_decode($other);
```

Comment:
mhh can we comment/explain why we need to do this? Also used in `additionalFailureDescription` mb put this in a priv commented method to avoid code duplication?